ament_cmake_gtest/ament_cmake_gtest-extras.cmake

```diff
@@ -87,6 +87,7 @@ macro(_ament_cmake_gtest_find_gtest)
         # mark gtest targets with EXCLUDE_FROM_ALL to only build
         # when tests are built which depend on them
         set_target_properties(gtest gtest_main PROPERTIES EXCLUDE_FROM_ALL 1)
+	target_compile_options(gtest PRIVATE -Wno-null-dereference)
```

Comment:
There are a couple of problems here.

The small problem is that the indentation is wrong.

The larger problem is that this won't work in a cross-platform way.  It *might* work on clang on macOS, but it definitely will not work on MSVC on Windows.  So we can't do it this way.  My preference would be to actually fix the problem in googletest itself with a patch to https://github.com/google/googletest.  If we can't get traction there, then a patch to https://github.com/ament/googletest would also work.

Which brings me to my last question.  We recently updated the Rolling version of Googletest to 1.10.0.  Is the issue fixed there?

Reply:
I apologize for the indentation error, I was careless.

Regarding the MSVC/macOS, I could check the compiler/OS before adding that flag.

I'm currently working on Foxy, I'll give it a shot in rolling. 

I'm using the docker ros:rolling, but the last version of gtest_vendor I can install is `1.8.9000-2focal.20200910.213617`
How could I try the latest version?

Reply:
> How could I try the latest version?

That's right, it was released after the latest Rolling sync.  The easiest thing to do is to enable the rolling testing repository, by changing your /etc/apt/sources.list.d/ros2-latest.list to look like:

```
deb http://packages.ros.org/ros2-testing/ubuntu focal main
```